extmod/nimble/modbluetooth_nimble.c

```diff
@@ -797,16 +798,16 @@ int mp_bluetooth_set_preferred_mtu(uint16_t mtu) {
 #if MICROPY_PY_BLUETOOTH_ENABLE_CENTRAL_MODE
 
 STATIC void gattc_on_data_available(uint8_t event, uint16_t conn_handle, uint16_t value_handle, const struct os_mbuf *om) {
-    size_t len = OS_MBUF_PKTLEN(om);
-    mp_uint_t atomic_state;
-    len = mp_bluetooth_gattc_on_data_available_start(event, conn_handle, value_handle, len, &atomic_state);
-    while (len > 0 && om != NULL) {
-        size_t n = MIN(om->om_len, len);
-        mp_bluetooth_gattc_on_data_available_chunk(OS_MBUF_DATA(om, const uint8_t *), n);
-        len -= n;
+    if (!om) {
```

Comment:
this case of `om==NULL` was not handled before, right?  is it ever possible for `om==NULL`?  should it raise a notify event with no data in such a case?

Reply:
On further investigation, it's not possible.

I've reworked this to no longer require adding the `offset` arg.